Update streamly to the latest commit

adithyaov opened the pull request:
This commit also bumps version of all other related packages

adithyaov committed:
Update streamly to the latest commit

This commit also bumps version of all other related packages

adithyaov commented on streaming-benchmarks.cabal:
```diff
@@ -96,14 +95,14 @@ benchmark bmarks
+    streamly            >= 0.7.0 && < 0.8,
     streaming           >= 0.1.4 && < 0.3,
-    machines            >= 0.6.0 && < 0.7,
+    machines            >= 0.7.0 && < 0.8,
```
Probably can keep this at `0.6.0`.

harendra-kumar reviewed (commented):
1. dlist can be bumped to < 1.1 and template-haskell to < 2.17
2. we can put bench-show/chart executable build under a flag if we do not want those dependencies to affect regular build.

adithyaov committed:
Add a development flag (dev) to the executable makecharts

This lets us bump the lts version of stack.
Diagrams universe that bench-show depends on is not supported by stackage yet.

adithyaov committed:
Update stack to the latest resolver

adithyaov committed:
Add a github CI that runs the build on linux

adithyaov committed:
Remove .travis.yml

adithyaov committed:
Remove additional stack files. Keep the CI story simple.

harendra-kumar committed:
Upgrade to streamly master branch

harendra-kumar committed:
Add fusion-plugin build flag

harendra-kumar committed:
Find "makecharts" from "charts" dir

harendra-kumar committed:
Update README with latest results

harendra-kumar committed:
Update streamly commit to latest on master branch
